stock_available_to_promise_release/models/stock_move.py

```diff
@@ -215,33 +215,50 @@ def _previous_promised_qty_sql_lateral_where(self):
             params["horizon"] = horizon_date
         return sql, params
 
-    def _previous_promised_qty_sql(self):
+    def _previous_promised_qty_sql(self, warehouse):
         """Lookup query for product promised qty in the same warehouse.
 
         Moves to consider are either already released or still be to released
         but not done yet. Each of them should fit the reservation horizon.
         """
         params = {"move_ids": tuple(self.ids)}
-        lateral_where, lateral_params = self._previous_promised_qty_sql_lateral_where()
+        lateral_where, lateral_params = self._previous_promised_qty_sql_lateral_where(
+            warehouse
+        )
         params.update(lateral_params)
         query = self._previous_promised_qty_sql_main_query().format(
             lateral_where=lateral_where
         )
         return query, params
 
+    def _group_by_warehouse(self):
+        return groupby(self, lambda m: m.warehouse_id)
```

Comment:
```suggestion
        return groupby(self, lambda m: m.warehouse_id or m.location_id._get_closest_warehouse())
```
Why not using this then?

Reply:
Seems legit as the warehouse is not required on the move

Reply:
We could still have moves MTS without warehouse set on them, so without this fallback value we won't include such moves in the computation of the previous promised qty.

Reply:
they are included if move has a location which is a child of warehouse.view_location_id

Reply:
My bad, it's the other way around:
If you want to compute the previous promised qty on a MTS move without warehouse, the current implementation will return 0, right?
That probably why you said we cannot release moves without warehouse with this change? (if so, that's problematic)

Reply:
Yes i will return 0, but there shouldn't be a mts without warehouse. The warehouse gets set by the stock.rule run https://github.com/odoo/odoo/blob/6b9b598fbb29d0bd22172dd346f9b36fae43e0bd/addons/stock/models/stock_rule.py#L321

Reply:
If we go this way, we need to explain it in the README

Reply:
Indeed, as `warehouse_id` is not a required field on the rule. We do have some MTS moves without warehouse for a customer, v14 as well (not  a lot but it happens).

Reply:
I can in fact re-add the closest warehouse. But it will not catch everything.
If there would be a rule without warehouse and the location is not located under a warehouse i will return 0. 
For v14 we would need stock_helper as dependency, for Version >= 16 we don't need it anymore because of https://github.com/odoo/odoo/pull/118128

Reply:
yes sure, but it is unfortunate to get a move to release with a location outside of any warehouse, it's not something we should support I think

Reply:
@mt-software-de @jbaudoux so what do we do about this? The fallback on closest warehouse hasn't been added.
For 16.0 I confirm that I could not found MTS moves without warehouse in one of our DB, so the current code works as expected for this version.

Reply:
For me it is fine, to merge without the fallback. On our project is isn't a problem.
In the most cases i think there is no MTS move which needs a release without a warehouse.